diff --git a/lib/models/blueprint.js b/lib/models/blueprint.js
index 7b557c5d725622c5b60b3a04952ee754b55b2c8e..51a166695cb5a06fe5a94ef00829373a81b1ef8d 100644
--- a/lib/models/blueprint.js
+++ b/lib/models/blueprint.js
@@ -1472,14 +1472,25 @@ Blueprint.load = function (blueprintPath, blueprintOptions) {
   if (fs.lstatSync(blueprintPath).isDirectory()) {
     let Constructor = Blueprint;
 
-    let constructorPath = path.resolve(blueprintPath, 'index.js');
-    if (Blueprint._existsSync(constructorPath)) {
-      const blueprintModule = require(constructorPath);
-
-      if (typeof blueprintModule === 'function') {
-        Constructor = blueprintModule;
-      } else {
-        Constructor = Blueprint.extend(blueprintModule);
+    let candidates = ['index.js', 'index.cjs', 'index.mjs'];
+
+    for (let candidateSuffix of candidates) {
+      let constructorPath = path.resolve(blueprintPath, candidateSuffix);
+
+      if (Blueprint._existsSync(constructorPath)) {
+        let blueprintModule = require(constructorPath);
+
+        if (candidateSuffix.endsWith('.mjs')) {
+          blueprintModule = blueprintModule.default;
+        }
+
+        if (typeof blueprintModule === 'function') {
+          Constructor = blueprintModule;
+        } else {
+          Constructor = Blueprint.extend(blueprintModule);
+        }
+
+        break;
       }
     }
 
